src/mcts/node.h

```diff
@@ -362,6 +363,11 @@ class EdgeAndNode {
     return numerator * GetP() / (1 + GetNStarted());
   }
 
+  float GetNewU(float numerator) const {
+    const float x = 1 + GetNStarted();
+    return numerator * GetP() * FastInvSqrt(x) / x;
```

Comment:
@borg323 could you take a look at this function and the referenced `FastInvSqrt` and check if it could be optimized further?

Reply:
isn't fastinvsqrt(x)/x just sqrt(x)?

Reply:
It’s equivalent to `1/(x * sqrt(x)) = 1/pow(x, 3./2.)`

Reply:
some guesses
1. (numerator * GetP()) * (FastInvSqrt(x) / x) may be faster on strict fp because it breaks a dependency
2. GetNStarted has a branch, maybe use a dummy node instead of nullptr?
3. rsqrtss should really be faster, it has latency of 5 and tp of 1 cpi

Reply:
The really fast answer is to use the awful sort of floating point bithacks that finvsqrt uses. Not saying that's a good idea though

Reply:
`return numerator * GetP() * FastInvSqrt(x * x * x);` should be slightly faster.

Reply:
http://quick-bench.com/u2AGvU2ZmWLY2bGqH4ykIpN2_IQ
![NPS with new U - random backend (4 runs per build)](https://user-images.githubusercontent.com/8037982/63189054-51e88900-c063-11e9-9f7a-481eca4ad2fe.png)
I used the newest MSVC compiler, intel i7-920
`lc0 benchmark -w bad-gyal-1.pb.gz --backend=random --threads=1 --smart-pruning-factor=0 --movetime=-1 --nodes=1000000 --new-u-enabled`
The last includes points 1, 2, 3 from my comment above. (in order of builds {nothing; 3; 2,3; 1,2,3}) I'm not sure if I patched correctly all the places that checked whether _node == nullptr (I did all local ones, but I don't know if the node can get 'spilled' outside)
If someone could retest on newer cpu it would be great
profiles before/after
![before](http://prntscr.com/otmbao)
![after](http://prntscr.com/otm9tt)
the `Actualize()` in edge iterator becomes the biggest problem, though it looks hard to improve without architectural changes

changes: https://github.com/kiudee/lc0/compare/new_u...Sopel97:opt_new_u

Reply:
This commit causes a reproducible segmentation fault when running new_u with `LogLiveStats` enabled.

Reply:
I look at this. Tried a few things. Can't get it to work without breaking other things.

Reply:
Thank you for looking into it @Sopel97.

@Tilps @borg323 The difference in performance (using the random backend) for Sopel97's branch is only ~7%, while being quite error prone due to the changes to the pointers. I think it would be better to have a separate pull request later for this performance optimization, which can be properly reviewed and tested. What is your opinion?

Reply:
7% is more than I expected. This is certainly worth a separate pr for review and discussion.